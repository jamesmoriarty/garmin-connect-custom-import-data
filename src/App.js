// import logo from './logo.svg';
import './App.css';
import Download from './download';

function App() {
  return (
    <div className="App">
      <Download />
    </div>
  );
}

export default App;
