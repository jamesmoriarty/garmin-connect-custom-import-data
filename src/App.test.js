import { render, screen } from '@testing-library/react';
import App from './App';

beforeAll(() => {
  jest.useFakeTimers('modern');
  jest.setSystemTime(new Date(2025, 3, 1));
});

afterAll(() => {
  jest.useRealTimers();
});

test('App', () => {
  const element = render(<App />).asFragment();

  expect(element).toMatchSnapshot();
});
