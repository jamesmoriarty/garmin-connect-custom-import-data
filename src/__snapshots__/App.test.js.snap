// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <form>
      <div>
        <label>
          Date:
          <input
            name="date"
            value="1/04/2025"
          />
        </label>
      </div>
      <div>
        <label>
          Time:
          <input
            name="time"
            value="00:00:00"
          />
        </label>
      </div>
      <div>
        <label>
          Age:
          <input
            name="age"
            value="37"
          />
        </label>
      </div>
      <div>
        <label>
          Height (cm):
          <input
            name="height"
            value="170"
          />
        </label>
      </div>
      <div>
        <label>
          Weight (kg):
          <input
            name="weight"
            value="75"
          />
        </label>
      </div>
      <div>
        <label>
          BMI:
          <input
            name="bmi"
            value="25"
          />
        </label>
      </div>
      <div>
        <label>
          Fat (%):
          <input
            name="fat"
            value="22"
          />
        </label>
      </div>
      <hr />
      <div>
        <label>
          Preview:
        </label>
        <pre>
          <code>
            Body
Date,Time,Weight,BMI,Fat
1/04/2025,00:00:00,75,25,22

          </code>
        </pre>
      </div>
      <a
        download="weight.csv"
        href="data:text/plain;base64,Qm9keQpEYXRlLFRpbWUsV2VpZ2h0LEJNSSxGYXQKMS8wNC8yMDI1LDAwOjAwOjAwLDc1LDI1LDIyCg=="
        role="button"
      >
        Download
      </a>
      <hr />
      <a
        class="secondary"
        href="https://connect.garmin.com/modern/import-data"
        rel="noreferrer"
        role="button"
        target="_blank"
      >
        Import Data
      </a>
      <hr />
    </form>
  </div>
</DocumentFragment>
`;
